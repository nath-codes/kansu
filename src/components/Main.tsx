import React from "react";
import styled from "styled-components";

interface Props {
  children: any;
}

export const Main = ({ children }: Props) => (
  <StyledMain>{children}</StyledMain>
);

const StyledMain = styled.main`
  position: relative;
`;
