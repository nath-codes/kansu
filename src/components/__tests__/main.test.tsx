import 'jest-styled-components'
import React from 'react'

import Main from '../main'
import { renderWithTheme } from './helpers'

describe('Main', () => {
  it('renders correctly', () => {
    let tree = renderWithTheme(
      <Main isMobileNavActive={false}>
        <div>foo</div>
      </Main>
    ).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderWithTheme(
      <Main isMobileNavActive={true}>
        <div>foo</div>
      </Main>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
