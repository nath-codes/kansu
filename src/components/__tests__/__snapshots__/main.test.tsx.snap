// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
.c1 {
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

.c0 {
  position: relative;
}

<main
  className="c0"
>
  <div
    className="c1"
    style={
      Object {
        "filter": "blur(0.75rem)",
      }
    }
  >
    0
  </div>
  <div>
    foo
  </div>
</main>
`;

exports[`Main renders correctly 2`] = `
.c1 {
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

.c0 {
  position: relative;
  -webkit-filter: blur(0.25rem);
  filter: blur(0.25rem);
}

<main
  className="c0"
>
  <div
    className="c1"
    style={
      Object {
        "filter": "blur(0.75rem)",
      }
    }
  >
    0
  </div>
  <div>
    foo
  </div>
</main>
`;
